tests/dotnet/UnitTests/ResourcePrefixTest.cs

```diff
@@ -0,0 +1,138 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+namespace Xamarin.Tests {
+	[TestFixture]
+	public class ResourcePrefixTest : TestBaseClass {
+		[Test]
+		public void ResourcePrefix_DefaultValues ()
+		{
+			// Arrange
+			var platforms = Configuration.GetAllPlatforms ();
+
+			foreach (var platform in platforms) {
+				// Act & Assert
+				var defaultValue = GetResourcePrefix (platform);
+				Assert.AreEqual ("Resources", defaultValue, $"Default value for {platform} should be 'Resources'");
+			}
+		}
+
+		[Test]
+		public void ResourcePrefix_AppBundleResourcePrefix ()
+		{
+			// Arrange
+			var platforms = Configuration.GetAllPlatforms ();
+			var customPrefix = "CustomResources";
+
+			foreach (var platform in platforms) {
+				// Act
+				var value = GetResourcePrefix (platform, ("AppBundleResourcePrefix", customPrefix));
+
+				// Assert
+				Assert.AreEqual (customPrefix, value, $"{platform}: AppBundleResourcePrefix should be used");
+			}
+		}
+
+		[Test]
+		public void ResourcePrefix_PlatformSpecific ()
+		{
+			// Arrange
+			var customPrefix = "CustomResources";
+
+			// Act & Assert
+
+			// iOS and tvOS use IPhoneResourcePrefix
+			var iOSValue = GetResourcePrefix ("ios", ("IPhoneResourcePrefix", customPrefix));
+			Assert.AreEqual (customPrefix, iOSValue, "iOS should use IPhoneResourcePrefix");
+
+			var tvOSValue = GetResourcePrefix ("tvos", ("IPhoneResourcePrefix", customPrefix));
+			Assert.AreEqual (customPrefix, tvOSValue, "tvOS should use IPhoneResourcePrefix");
+
+			// Mac Catalyst uses IPhoneResourcePrefix
+			var macCatalystValue = GetResourcePrefix ("maccatalyst", ("IPhoneResourcePrefix", customPrefix));
+			Assert.AreEqual (customPrefix, macCatalystValue, "Mac Catalyst should use IPhoneResourcePrefix");
+
+			// macOS can use either XamMacResourcePrefix or MonoMacResourcePrefix
+			var macOSXamValue = GetResourcePrefix ("macos", ("XamMacResourcePrefix", customPrefix));
+			Assert.AreEqual (customPrefix, macOSXamValue, "macOS should use XamMacResourcePrefix");
+
+			var macOSMonoValue = GetResourcePrefix ("macos", ("MonoMacResourcePrefix", customPrefix));
+			Assert.AreEqual (customPrefix, macOSMonoValue, "macOS should use MonoMacResourcePrefix");
+		}
+
+		[Test]
+		public void ResourcePrefix_Precedence ()
+		{
+			// Arrange
+			var appBundlePrefix = "AppPrefix";
+			var platformPrefix = "PlatformPrefix";
+
+			// Act & Assert
+
+			// iOS - AppBundleResourcePrefix should take precedence over IPhoneResourcePrefix
+			var iOSValue = GetResourcePrefix ("ios",
```

Comment:
Use `Configuration.include_ios` here too, and the equivalent for the other platforms.